tests/bdp_spark_pi_submit_test.erl

```diff
@@ -0,0 +1,155 @@
+    ok = bdp_util:wait_services(Node2, {[], [?SPARK_MASTER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_NAME]}),
+
+    lager:info("Service ~p (~s) added", [?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE]),
+    %start master and worker
+    ok = bdp_util:start_seervice(Node1, Node1, ?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE),
+    ok = bdp_util:wait_services(Node1, {[?SPARK_MASTER_SERVICE_NAME], [?SPARK_MASTER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_NAME]}),
+    ok = bdp_util:start_seervice(Node2, Node2, ?SPARK_WORKER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE),
+    ok = bdp_util:wait_services(Node2, {[?SPARK_MASTER_SERVICE_NAME,?SPARK_WORKER_SERVICE_NAME], [?SPARK_MASTER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_NAME]}),
+
+    lager:info("Waiting 10 seconds..."),
+    timer:sleep(10000),
+
+    %check running services
+    {Run,Ava} = bdp_util:get_services(Node1),
+
+    MyServices = ["my-spark-master","my-spark-worker"],
+    ?assert(Run == MyServices),
+    ?assert(Ava == MyServices),
+    lager:info("~s",[MyServices]),
+    lager:info("Running: ~s", [Run]),
+    lager:info("Available: ~s", [Ava]),
+
+    %%run spark job pi.py
+    SparkJobSubmit1 = "./lib/data_platform-1/priv/spark-master/bin/spark-submit --master ",
+    SparkJobSubmit2 = " ./lib/data_platform-1/priv/spark-master/examples/src/main/python/pi.py 100",
+    SparkJobSubmit3 = SparkJobSubmit1++SparkMasterAddr++SparkJobSubmit2,
+    Results = rpc:call(Node2,os,cmd,[SparkJobSubmit3]),
+    lager:info("Spark Job Results: ~s", [Results]),
+
+    %Test if the spark job submission worked
+    ?assert(string:str(Results,"Pi is roughly") > 0),
+
+    %Assert proper cluster execution of spark job
+    WgetCall = "wget "++SparkMasterIP++":8080",
+    lager:info("wget call: ~s",[WgetCall]),
+    rpc:call(Node1,os,cmd,[WgetCall]),
+    {ok, File} = rpc:call(Node1,file,read_file,["index.html"]),
+    Content = unicode:characters_to_list(File),
+    %assert master and worker see each other
+    TestString1 = "<li><strong>Workers:</strong> 1</li>",
+    ?assert(string:str(Content,TestString1) > 0),
+    %assert job execution in cluster
+    TestString2 = "<li><strong>Applications:</strong>
+                0 Running,
+                1 Completed </li>",
+    ?assert(string:str(Content,TestString2) > 0),
+
+    pass.
+
+
+-spec get_routed_interfaces(node()) -> [{Iface::string(), inet:ip_address()}].
+%% @private
+get_routed_interfaces(Node) ->
+    case rpc:call(Node, inet, getifaddrs, []) of
+        {ok, Ifaces} ->
+            lists:filtermap(
+              fun({Iface, Details}) ->
+                      case is_routed_addr(Details) of
+                          undefined ->
+                              false;
+                          Addr ->
+                              {true, {Iface, Addr}}
+                      end
+              end,
+              Ifaces);
+        {error, PosixCode} ->
```

Comment:
fail fast, if we cannot reach and run inet:getifaddrs on a node, the test results should be a fail due to improper setup. Letting this failure continue makes finding the issue harder.